<a href="https://colab.research.google.com/github/RivasAlberto/PowerBI/blob/main/Desafio05_Python.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Desafío 05: Proyecto Avanzado: Dominio de manipulación, limpieza  y transformación de Datos con Python**  
_Realizado por: Jorge Alberto Rivas Sánchez_  
- **Curso:** Data Science 98  
- **Módulo:** Programación en Python para el Análisis de Datos  

Este proyecto aborda un desafío práctico de manipulación y análisis de datos mediante Python y pandas, enfocado en limpiar, transformar y obtener insights clave de diferentes datasets. Este trabajo pone en evidencia competencias en limpieza de datos, creación de tablas pivote, manejo de valores nulos y duplicados, y consolidación de información para generar soluciones efectivas en entornos empresariales.

**"Habilidades destacada**s: análisis de datos, limpieza y transformación de datasets, creación de tablas pivote, y fusión de DataFrames con Python."

# **Desafío: Manipulación y Transformación de Datos (Parte 1)**
Este desafío práctico pone a prueba nuestras habilidades en la manipulación de datos utilizando Python y pandas. A partir de diferentes archivos, llevaremos a cabo tareas avanzadas para consolidar, limpiar y analizar los datos.

## **Archivos utilizados:**

**incidents.pkl**: Contiene información sobre incidentes registrados.

**officers.pkl**: Datos relacionados con oficiales involucrados.

**subjects.pkl**: Información detallada de los sujetos involucrados.

**Cleaned_DS_Jobs.csv**: Dataset relacionado con ofertas laborales y sus características clave.


In [ ]:
# Importar las bibliotecas necesarias
import pandas as pd

# Cargar el archivo 'incidents.pkl'
incidents_df = pd.read_pickle('incidents.pkl')
# Mostrar las primeras filas del DataFrame 'incidents_df' para verificar la carga de datos
print("Incidents DataFrame:")
print(incidents_df.head())

# Cargar el archivo 'officers.pkl'
officers_df = pd.read_pickle('officers.pkl')
# Mostrar las primeras filas del DataFrame 'officers_df' para verificar la carga de datos
print("\nOfficers DataFrame:")
print(officers_df.head())

# Cargar el archivo 'subjects.pkl'
subjects_df = pd.read_pickle('subjects.pkl')
# Mostrar las primeras filas del DataFrame 'subjects_df' para verificar la carga de datos
print("\nSubjects DataFrame:")
print(subjects_df.head())




Incidents DataFrame:
  case_number        date                  location subject_statuses  \
0      44523A  2013-02-23     3000 Chihuahua Street          Injured   
1     121982X  2010-05-03  1300 N. Munger Boulevard          Injured   
2     605484T  2007-08-12   200 S. Stemmons Freeway            Other   
3     384832T  2007-05-26           7900 S. Loop 12   Shoot and Miss   
4     244659R  2006-04-03        6512 South Loop 12          Injured   

  subject_weapon                                           subjects  \
0        Handgun                                   Curry, James L/M   
1        Handgun                                Chavez, Gabriel L/M   
2        Shotgun                                  Salinas, Nick L/M   
3        Unarmed  Smith, James B/M; Dews, Antonio B/M; Spearman,...   
4          Hands                                 Watkins, Caleb B/M   

   subject_count                                   officers  officer_count  \
0              1  Patino, Michael L/M; Fi

# **Unir los 3 dataframes creados en un solo**

Integramos los tres DataFrames en una única tabla consolidada. Esto permite analizar la relación entre incidentes, oficiales y sujetos desde una perspectiva unificada.

---



In [ ]:
# Unir officers_df y subjects_df usando la columna 'case_number' y añadiendo sufijos para columnas comunes
df_merge1 = pd.merge(officers_df, subjects_df, on='case_number', suffixes=('_officer', '_subject'))

# Mostrar las primeras filas del DataFrame 'df_merge1' para verificar la unión
print("df_merge1 (officers + subjects):")
print(df_merge1.head())

# Unir df_merge1 con incidents_df usando la columna 'case_number' y añadiendo sufijos para columnas comunes
final_df = pd.merge(df_merge1, incidents_df, on='case_number', suffixes=('', '_incident'))

# Mostrar las primeras filas del DataFrame 'final_df' para verificar la unión
print("\nfinal_df (df_merge1 + incidents):")
print(final_df.head())



df_merge1 (officers + subjects):
  case_number race_officer gender_officer last_name_officer  \
0      44523A            L              M            Patino   
1      44523A            W              M         Fillingim   
2     121982X            L              M           Padilla   
3     605484T            W              M            Poston   
4     384832T            B              M             Mondy   

  first_name_officer full_name_officer race_subject gender_subject  \
0            Michael   Patino, Michael            L              M   
1              Brian  Fillingim, Brian            L              M   
2            Gilbert  Padilla, Gilbert            L              M   
3              Jerry     Poston, Jerry            L              M   
4            Michael    Mondy, Michael            B              M   

  last_name_subject first_name_subject full_name_subject  
0             Curry              James      Curry, James  
1             Curry              James      Curry

# **Verificación de filas duplicadas.**
Realizamos una inspección exhaustiva para identificar y eliminar duplicados, asegurando la integridad de los datos y evitando resultados distorsionados en el análisis posterior.

In [ ]:
# Verificar si hay filas duplicadas en 'final_df'
duplicated_rows = final_df.duplicated()

# Mostrar el número de filas duplicadas
print(f"Número de filas duplicadas: {duplicated_rows.sum()}")

# Eliminar las filas duplicadas
final_df_cleaned = final_df.drop_duplicates()

# Mostrar el número de filas después de eliminar duplicados para verificar
print(f"Número de filas después de eliminar duplicados: {final_df_cleaned.shape[0]}")



Número de filas duplicadas: 0
Número de filas después de eliminar duplicados: 377


# **Conteo de sujetos de género F.**
Determinamos cuántos sujetos del género femenino están presentes en los datos, lo cual puede ser útil para detectar patrones específicos.

In [ ]:
# Contar cuántos sujetos de género F hay en el DataFrame 'final_df_cleaned'
gender_counts = final_df_cleaned['gender_subject'].value_counts()

# Mostrar el número de sujetos de género F
num_female_subjects = gender_counts.get('F', 0)
print(f"Número de sujetos de género F: {num_female_subjects}")


Número de sujetos de género F: 9


# **Número de casos con sospechosas mujeres.**
Filtramos y contamos los casos en los que al menos una mujer figura como sospechosa, utilizando herramientas como unique() para identificar estos eventos de manera precisa.

In [ ]:
# Filtrar las filas donde el género del sujeto es 'F'
female_subjects = final_df_cleaned[final_df_cleaned['gender_subject'] == 'F']

# Obtener los valores únicos de 'case_number' donde hay al menos una sospechosa mujer
unique_case_numbers_with_female_subjects = female_subjects['case_number'].unique()

# Contar el número de casos únicos
num_cases_with_female_subjects = len(unique_case_numbers_with_female_subjects)

# Mostrar el número de casos únicos con al menos una sospechosa mujer
print(f"Número de casos con al menos una sospechosa mujer: {num_cases_with_female_subjects}")


Número de casos con al menos una sospechosa mujer: 7


# **Generaración tabla pivote.**

Creamos una tabla pivote para analizar la relación entre el género del oficial y el género de los sujetos involucrados. Esta visualización ayuda a detectar tendencias o patrones de género en los incidentes registrados.



In [ ]:
# Crear la tabla pivote con el género del oficial en las filas y el género del sujeto en las columnas
#Usamos la funcion pivot_table de panda
pivot_table = final_df_cleaned.pivot_table(
    values='case_number',
    # Especifica que gender_officer debe estar en las filas
    index='gender_officer',
    # Especifica que gender_subject debe estar en las columnas
    columns='gender_subject',
    #cuenta el número de case_number en cada combinación de género.
    aggfunc='count',
    #asegura que las celdas vacías se llenen con 0.
    fill_value=0
)

# Mostrar la tabla pivote resultante
print("Tabla Pivote - Género del Oficial vs. Género del Sujeto:")
print(pivot_table)


Tabla Pivote - Género del Oficial vs. Género del Sujeto:
gender_subject  F    M
gender_officer        
F               2   18
M               7  350


# **Interpretación del resultado de la vista de la tabla pivote.**
Los valores de la tabla pivote nos brindan una perspectiva clara sobre la interacción entre oficiales y sujetos de diferentes géneros. Esto permite identificar posibles tendencias o sesgos que podrían ser relevantes para investigaciones o toma de decisiones.


# **Interpretación.**
**En cuanto a a Distribución de Géneros:**

Oficiales Femeninos (F): Participaron en casos con 2 sujetos femeninos y 18 sujetos masculinos.

Oficiales Masculinos (M): Participaron en casos con 7 sujetos femeninos y 350 sujetos masculinos.

**Observaciones:**
1. Los oficiales masculinos (M) tienen una participación significativamente mayor en casos con sujetos masculinos (M), sumando un total 350 casos. Esto sugiere una mayor asignación o predominio de oficiales masculinos en situaciones que involucran sujetos masculinos.

2. Los oficiales femeninos (F) están menos representados en ambos casos, se registran tan solo  2 casos con sujetos femeninos y 18 con sujetos masculinos. Esto puede ser un  reflejo de la proporción general de oficiales femeninos en la fuerza policial o una tendencia en la asignación de casos.

**Relaciones entre géneros:**
Es interesante encontrar que hay más interacción entre oficiales masculinos y sujetos masculinos. Esto podría estar relacionado con la composición demográfica de la fuerza policial y los sujetos involucrados en los incidentes.

La presencia de oficiales femeninos en casos con sujetos femeninos es muy baja (solo 2 casos), lo cual podría merecer una investigación adicional para entender las razones detrás de esta distribución.

La distribución de género entre los oficiales y los sujetos involucrados puede reflejar ciertas tendencias organizacionales y sociales.

Potencial de Sesgo en la Asignación de Casos: Si ciertos tipos de casos se asignan predominantemente a oficiales masculinos, puede haber un sesgo organizacional en la asignación de trabajo que necesita ser revisado.

**Acciones recomendadas:**
**Revisión de Políticas de Asignación**: Analizar las políticas de asignación de casos para identificar posibles sesgos y trabajar hacia una distribución más equitativa.

**Programas de Mentoría y Entrenamiento**: Implementar programas específicos para apoyar el desarrollo profesional de oficiales femeninas y fomentar un ambiente de trabajo inclusivo.

**Recolección de Datos Más Detallada**: Continuar recolectando y analizando datos con mayor detalle para entender mejor las dinámicas de género dentro de la fuerza policial y en los incidentes reportados.

**Conclusión:**
La tabla pivote nos ha brindado una valiosa perspectiva sobre la distribución de género y su posible impacto en la dinámica de trabajo dentro de la fuerza policial. Al entender estas interacciones, se pueden tomar decisiones informadas para mejorar la equidad y eficacia en el trabajo policia

# **SEGUNDA PARTE DEL DESAFIO**
## **SOBRE EL ARCHIVO Cleaned_DS_Jobs.csv**


En esta sección, trabajamos con un dataset relacionado con ofertas laborales en el ámbito de la ciencia de datos, aplicando técnicas avanzadas de limpieza y transformación.



# **Cargar el archivo y crear un DataFrame.**
Se carga el archivo Cleaned_DS_Jobs.csv en un DataFrame para iniciar el análisis, asegurando una correcta estructura para la manipulación de datos.

In [ ]:
# Importar la biblioteca necesaria
import pandas as pd

# Cargar el archivo 'Cleaned_DS_Jobs.csv' en un DataFrame
ds_jobs_df = pd.read_csv('Cleaned_DS_Jobs.csv')

# Mostrar las primeras filas del DataFrame para verificar la carga de datos
print("DataFrame 'ds_jobs_df':")
print(ds_jobs_df.head())
#ds_jobs_df.head(60)


DataFrame 'ds_jobs_df':
           Job Title Salary Estimate  \
0  Sr Data Scientist        137-171    
1     Data Scientist        137-171    
2     Data Scientist        137-171    
3     Data Scientist        137-171    
4     Data Scientist        137-171    

                                     Job Description  Rating  \
0  Description\n\nThe Senior Data Scientist is re...     3.1   
1  Secure our Nation, Ignite your Future\n\nJoin ...     4.2   
2  Overview\n\n\nAnalysis Group is one of the lar...     3.8   
3  JOB DESCRIPTION:\n\nDo you have a passion for ...     3.5   
4  Data Scientist\nAffinity Solutions / Marketing...     2.9   

         Company Name       Location            Headquarters  \
0         Healthfirst   New York, NY            New York, NY   
1             ManTech  Chantilly, VA             Herndon, VA   
2      Analysis Group     Boston, MA              Boston, MA   
3             INFICON     Newton, MA  Bad Ragaz, Switzerland   
4  Affinity Solutions   New Yo

# **Manejo de valores nulos**.
Identificamos y reemplazamos valores inconsistentes o considerados como nulos ("na", "NA", -1, "0", "-1", "null", "n/a", "N/A", "NULL") por NaN. Esto nos permite limpiar el dataset y trabajar únicamente con datos relevantes.

In [ ]:
# Importar las bibliotecas necesarias
import pandas as pd
import numpy as np

# Lista de valores a considerar como nulos
null_values = ["na", "NA", -1, "0", "-1", "null", "n/a", "N/A", "NULL"]

# Cargar el archivo 'Cleaned_DS_Jobs.csv' en un DataFrame
ds_jobs_df = pd.read_csv('Cleaned_DS_Jobs.csv')

# Reemplazar los valores indicados por np.nan
ds_jobs_df.replace(null_values, np.nan, inplace=True)

# Mostrar las primeras filas del DataFrame para verificar la carga y la sustitución de valores nulos
print("DataFrame 'ds_jobs_df' con valores nulos reemplazados:")
print(ds_jobs_df.head())


DataFrame 'ds_jobs_df' con valores nulos reemplazados:
           Job Title Salary Estimate  \
0  Sr Data Scientist        137-171    
1     Data Scientist        137-171    
2     Data Scientist        137-171    
3     Data Scientist        137-171    
4     Data Scientist        137-171    

                                     Job Description  Rating  \
0  Description\n\nThe Senior Data Scientist is re...     3.1   
1  Secure our Nation, Ignite your Future\n\nJoin ...     4.2   
2  Overview\n\n\nAnalysis Group is one of the lar...     3.8   
3  JOB DESCRIPTION:\n\nDo you have a passion for ...     3.5   
4  Data Scientist\nAffinity Solutions / Marketing...     2.9   

         Company Name       Location            Headquarters  \
0         Healthfirst   New York, NY            New York, NY   
1             ManTech  Chantilly, VA             Herndon, VA   
2      Analysis Group     Boston, MA              Boston, MA   
3             INFICON     Newton, MA  Bad Ragaz, Switzerland   

# **Eliminación de filas con datos faltantes**.
Mediante el método .dropna(), eliminamos todas las filas incompletas, garantizando que el análisis posterior se base en datos confiables y completos.


In [ ]:
print(ds_jobs_df.head())

# Eliminar todas las filas con datos faltantes
ds_jobs_df_cleaned = ds_jobs_df.dropna()

# Mostrar las primeras filas del DataFrame limpio para verificar la eliminación de filas con datos faltantes
print("DataFrame 'ds_jobs_df_cleaned' sin datos faltantes:")
print(ds_jobs_df_cleaned.head())

# Mostrar el tamaño del DataFrame antes y después de eliminar las filas con datos faltantes para comparar
print(f"Tamaño original del DataFrame: {ds_jobs_df.shape}")
print(f"Tamaño del DataFrame limpio: {ds_jobs_df_cleaned.shape}")
ds_jobs_df_cleaned.head()


           Job Title Salary Estimate  \
0  Sr Data Scientist        137-171    
1     Data Scientist        137-171    
2     Data Scientist        137-171    
3     Data Scientist        137-171    
4     Data Scientist        137-171    

                                     Job Description  Rating  \
0  Description\n\nThe Senior Data Scientist is re...     3.1   
1  Secure our Nation, Ignite your Future\n\nJoin ...     4.2   
2  Overview\n\n\nAnalysis Group is one of the lar...     3.8   
3  JOB DESCRIPTION:\n\nDo you have a passion for ...     3.5   
4  Data Scientist\nAffinity Solutions / Marketing...     2.9   

         Company Name       Location            Headquarters  \
0         Healthfirst   New York, NY            New York, NY   
1             ManTech  Chantilly, VA             Herndon, VA   
2      Analysis Group     Boston, MA              Boston, MA   
3             INFICON     Newton, MA  Bad Ragaz, Switzerland   
4  Affinity Solutions   New York, NY            New Yo

Job Title Salary Estimate  \
0                              Sr Data Scientist        137-171    
32  Senior Research Statistician- Data Scientist         75-131    
38                 Senior Analyst/Data Scientist         75-131    
45                         Senior Data Scientist         75-131    
54                         Senior Data Scientist         75-131    

                                      Job Description  Rating  \
0   Description\n\nThe Senior Data Scientist is re...     3.1   
32  Acuity is seeking a Senior Research Statistici...     4.8   
38  At Edmunds were driven to make car buying easi...     3.4   
45  Klaviyo is looking for Senior Data Scientists ...     4.8   
54  Benson Hill empowers innovators to develop mor...     3.5   

        Company Name          Location      Headquarters  \
0        Healthfirst      New York, NY      New York, NY   
32  Acuity Insurance     Sheboygan, WI     Sheboygan, WI   
38       Edmunds.com  Santa Monica, CA  Santa Monica, CA   
45           Klaviyo        Boston, MA        Boston, MA   
54       Benson Hill   Saint Louis, MO   Saint Louis, MO   

                      Size       Type of ownership  \
0   1001 to 5000 employees  Nonprofit Organization   
32  1001 to 5000 employees       Company - Private   
38   501 to 1000 employees       Company - Private   
45    201 to 500 employees       Company - Private   
54    201 to 500 employees       Company - Private   

                        Industry  ... company_age python  excel  hadoop  \
0             Insurance Carriers  ...        27.0      0      0       0   
32            Insurance Carriers  ...        95.0      0      0       0   
38                      Internet  ...        54.0      1      1       0   
45  Computer Hardware & Software  ...         8.0      0      0       0   
54     Biotech & Pharmaceuticals  ...         8.0      1      1       0   

    spark aws  tableau  big_data        job_simp  seniority  
0       0   1        0         0  data scientist     senior  
32      0   0        0         0  data scientist     senior  
38      0   1        1         0  data scientist     senior  
45      0   0        0         0  data scientist     senior  
54      0   0        0         1  data scientist     senior  

[5 rows x 27 columns]

# **Generación de columnas: Salario Estimado Mínimo y Máximo.**
A partir de la columna "Salary Estimate", creamos dos nuevas columnas que representan los límites mínimo y máximo del salario estimado. Esto nos permite un análisis más detallado de las oportunidades laborales.

In [ ]:
# Importar las bibliotecas necesarias
import pandas as pd
import numpy as np

# Definir una función para extraer el salario estimado mínimo
def get_min_salary(salary):
    try:
        # Validar que el salario no sea nulo y contenga un rango separado por '-'
        if pd.isnull(salary) or '-' not in salary:
            return np.nan
        return int(salary.split('-')[0].strip('$K').replace(',', '').strip())
    except Exception as e:
        print(f"Error al procesar salario mínimo en: {salary}. Error: {e}")
        return np.nan

# Definir una función para extraer el salario estimado máximo
def get_max_salary(salary):
    try:
        # Validar que el salario no sea nulo y contenga un rango separado por '-'
        if pd.isnull(salary) or '-' not in salary:
            return np.nan
        return int(salary.split('-')[1].strip('K').strip('$').replace(',', '').strip())
    except Exception as e:
        print(f"Error al procesar salario máximo en: {salary}. Error: {e}")
        return np.nan

# Filtrar filas donde 'Salary Estimate' no esté vacío o nulo antes de aplicar las funciones
ds_jobs_df_cleaned = ds_jobs_df_cleaned[ds_jobs_df_cleaned['Salary Estimate'].notna()]

# Aplicar las funciones para calcular los salarios mínimos y máximos
ds_jobs_df_cleaned['Salario Estimado Mínimo'] = ds_jobs_df_cleaned['Salary Estimate'].apply(get_min_salary)
ds_jobs_df_cleaned['Salario Estimado Máximo'] = ds_jobs_df_cleaned['Salary Estimate'].apply(get_max_salary)

# Restablecer los índices del DataFrame
ds_jobs_df_cleaned.reset_index(drop=True, inplace=True)

# Mostrar las primeras filas del DataFrame para verificar las nuevas columnas
print("DataFrame con Salario Estimado Mínimo y Máximo, e índices corregidos:")
print(ds_jobs_df_cleaned[['Salary Estimate', 'Salario Estimado Mínimo', 'Salario Estimado Máximo']].head())




DataFrame con Salario Estimado Mínimo y Máximo, e índices corregidos:
  Salary Estimate  Salario Estimado Mínimo  Salario Estimado Máximo
0        137-171                       137                      171
1         75-131                        75                      131
2         75-131                        75                      131
3         75-131                        75                      131
4         75-131                        75                      131


# **Recodificación de la columna "Size"**
La columna "Size" se recodifica para ofrecer una descripción más comprensible y amigable sobre el tamaño de las empresas, utilizando los siguientes valores:

*   51 to 200 employees → Pequeñas Grandes Empresas
*   201 to 500 employees → Pequeñas Empresas
*   501 to 1,000 employees → Microempresas
*   1,001 to 5,000 employees → Medianas Empresas
*   5,001 to 10,000 employees → Grandes Empresas
*   10,000+ employees → Mega Empresas
*   Unknown → Empresas sin Información










In [ ]:
# Definir el diccionario de reemplazo basado en la tabla proporcionada
size_replacement = {
    "10000+ employees": "Mega Empresas",
    "5001 to 10000 employees": "Grandes Empresas",
    "1001 to 5000 employees": "Medianas Empresas",
    "201 to 500 employees": "Pequeñas Empresas",
    "51 to 200 employees": "Pequeñas Grandes Empresas",
    "501 to 1000 employees": "Microempresas",
    "Unknown": "Empresas sin Información"
}

# Reemplazar los valores en la columna 'Size' utilizando el diccionario con .loc
ds_jobs_df_cleaned.loc[:, 'Size'] = ds_jobs_df_cleaned['Size'].replace(size_replacement)

# Mostrar las primeras filas del DataFrame para verificar la recodificación
print("DataFrame después de recodificar la columna 'Size':")
print(ds_jobs_df_cleaned[['Size']].head(25))



DataFrame después de recodificar la columna 'Size':
                         Size
0           Medianas Empresas
1           Medianas Empresas
2               Microempresas
3           Pequeñas Empresas
4           Pequeñas Empresas
5            Grandes Empresas
6            Grandes Empresas
7   Pequeñas Grandes Empresas
8            Grandes Empresas
9               Microempresas
10              Mega Empresas
11  Pequeñas Grandes Empresas
12              Microempresas
13           Grandes Empresas
14           Grandes Empresas
15          Pequeñas Empresas
16          Pequeñas Empresas
17  Pequeñas Grandes Empresas
18              Mega Empresas
19           Grandes Empresas
20           Grandes Empresas
21          Medianas Empresas
22          Pequeñas Empresas
23              Microempresas
24          Pequeñas Empresas


# **Generación de tabla pivote.**
Creamos una tabla pivote para mostrar la media del salario estimado mínimo y máximo por tamaño de empresa. Esto proporciona una visión clara de cómo varían las oportunidades laborales según la escala organizacional.

In [ ]:
# Crear la tabla pivote
pivot_table_salaries = pd.pivot_table(
    ds_jobs_df_cleaned,
    values=['Salario Estimado Mínimo', 'Salario Estimado Máximo'],
    index='Size',
    aggfunc='mean'
)

# Mostrar la tabla pivote resultante
print("Tabla Pivote - Media del Salario Estimado por Tamaño de Empresa:")
print(pivot_table_salaries)


Tabla Pivote - Media del Salario Estimado por Tamaño de Empresa:
                           Salario Estimado Máximo  Salario Estimado Mínimo
Size                                                                       
Empresas sin Información                110.500000                73.000000
Grandes Empresas                        138.875000                92.125000
Medianas Empresas                       137.461538                93.923077
Mega Empresas                           151.111111                97.888889
Microempresas                           146.235294               100.176471
Pequeñas Empresas                       141.142857                93.571429
Pequeñas Grandes Empresas               137.666667               100.666667


# **Análisis de los datos de la tabla pivote de salarios por tipo de empresas.**
**Observaciones sobre el salario:**

**1. Salario Estimado Máximo:**

**Mega Empresas**: Con un salario estimado máximo promedio de 151.11.  Esto puede deberse a su capacidad financiera robusta y a su  mayor necesidad de atraer talento altamente calificado y posiblemente brindar beneficios adicionales que complementan el salario base.

**Microempresas:** Sorprendentemente, las microempresas también ofrecen salarios máximos competitivos (146.23). Esto sugiere que algunas microempresas, especialmente en sectores tecnológicos o startups, están dispuestas a ofrecer salarios altos para atraer talento clave y competir con las grandes corporaciones.

**2. Salario Estimado Mínimo:**

**Pequeñas Grandes Empresas y Microempresa**s: Estas categorías muestran salarios estimados mínimos altos (100.67 y 100.18 respectivamente). Esto puede reflejar una estrategia para ofrecer un salario mínimo atractivo que retenga a los empleados desde el principio.

**Mega Empresas:** Aunque el salario mínimo (97.88) es competitivo, las mega empresas tienden a ofrecer una gama más amplia de beneficios no salariales, como planes de jubilación, opciones de acciones y programas de bienestar, que pueden complementar el salario base. Lamentablemente en la data proporcionada no hay información al respecto.

**Observaciones sobre estrategias de las empresas**

Las grandes empresas pueden ofrecer paquetes de compensación más variados que van más allá del salario, incluyendo beneficios como seguro médico, planes de desarrollo profesional, y oportunidades de promoción interna.

Las pequeñas y medianas empresas para competir con las grandes empresas, pueden estar ofreciendo salarios más altos o paquetes más atractivos desde el principio para retener talento.

Las mega empresas pueden confiar en su reputación y beneficios adicionales para atraer y retener empleados, aunque sus salarios mínimos no sean los más altos.


## **Recomendaciones**
**Para Pequeñas Empresas:**

Hacer énfasis en Beneficios Adicionales: Ofrecer beneficios adicionales que no sean solo salariales, como flexibilidad laboral, desarrollo profesional y un ambiente de trabajo atractivo.

**Para Mega Empresas:**

Revisión de Salarios Mínimos: Aunque tienen otras ventajas, se sugiere ajustar los salarios mínimos para ser más competitivos y agilizar el atraer talentos que buscan una mejor compensación inicial.

**Para Todas las Empresas:**

Benchmarking: Realizar un benchmarking regular del mercado salarial para asegurarse de que las ofertas son competitivas y así atraer al talento laboral.

## **Conclusión:**
Este análisis revela cómo diferentes tamaños de empresas manejan sus estrategias salariales para atraer y retener talento. Mientras que las mega empresas ofrecen atractivos paquetes generales, las micro y pequeñas empresas pueden competir ofreciendo salarios iniciales más altos.
